tests/contrib/django/test_django.py

```diff
@@ -1460,3 +1463,29 @@ def test_template_view_patching():
     # Patch via `as_view()`
     TemplateView.as_view()
     assert "dispatch" not in vars(TemplateView)
+
+
+class _HttpRequest(django.http.HttpRequest):
+    @property
+    def scheme(self):
+        return b"http"
+
+
+@pytest.mark.parametrize(
+    "request_cls,request_path_encode,http_host_encode",
+    itertools.product([django.http.HttpRequest, _HttpRequest], [False, True], [False, True],),
+)
+def test_helper_get_request_uri(request_cls, request_path_encode, http_host_encode):
+    def encode(x, y):
```

Comment:
I'd be in favour of dropping this helper to avoid the indirection with request_path_encode, http_host_encode. Being more verbose here will help readability a ton.